pkg/scheduler/core/equivalence/eqivalence.go

```diff
@@ -150,6 +140,23 @@ func (c *Cache) InvalidateCachedPredicateItemForPodAdd(pod *v1.Pod, nodeName str
 		}
 	}
 	c.InvalidatePredicatesOnNode(nodeName, invalidPredicates)
+
+	// MatchInterPodAffinity: we assume scheduler can make sure newly bound pod
+	// will not break the existing inter pod affinity. So normally we does not need to
+	// invalidate MatchInterPodAffinity when pod added.
+	//
+	// NOTE: assumptions above will not stand when we implemented features like
+	// RequiredDuringSchedulingRequiredDuringExecutioc.
+	//
+	// But the exception is: pod has anti-affinity to itself. This is common in a ReplicaSet,
+	// so we have to invalidate MAtachInterPodAffinity in this case.
+	//
+	// TODO(harry): Should consider a more fine-grained way in the future, especially
+	// if eCache can be merged to predicates pkg.
+	invalidAntiAffinityPred := sets.NewString(predicates.MatchInterPodAffinityPred)
+	if pod.Spec.Affinity != nil && pod.Spec.Affinity.PodAntiAffinity != nil {
+		c.InvalidatePredicates(invalidAntiAffinityPred)
```

Comment:
If we remove anti-affinity predicates when pod added, when will cache be hit?

Reply:
Sadly, no :( 

That's why there's a TODO: we will invalidate this predicate on pod add
1. for nodes in the same failure domain only
2. for pods with self anti-affinity only

But I would prefer to implement this after we merge eCache into predicates (discussed in #67241).

Reply:
can we skip it?

Reply:
Instead of adding exception for a specific predicate, I would prefer to combine "cheapPredicates" logic here.

Reply:
We already have one in #67241, that's why I propose to combine these two. Thus the predicates will be classified by the reason why they are skipped.